### ISSUE TITLE
cmd/create-account can't connect global.database

### ISSUE THREAD

**Issue by nakoo** [matrix-org/dendrite#2449]
### Background information

- **Dendrite version or git SHA**: 0.8.4
- **Monolith or Polylith?**: Monolith
- **SQLite3 or Postgres?**: Postgres
- **Running in Docker?**: Yes
- **`go version`**: -
- **Client used (if applicable)**: Elemnt Web

### Description

 - **What** is the problem:

Since `create-account` command only checks `UserAPI.AccountDatabase`, `global.database` that included in v0.8.3 for monolith can't be connected.

 - **Who** is affected:
Who uses `create-account` command

 - **How** is this bug manifesting:
```
FATA[0004] Failed to connect to the database error="dial tcp 127.0.0.1:5432: connect: connection refused"
```

### Steps to reproduce

- run `create-account` command

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from server logs, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.

Alternatively, please send logs to @kegan:matrix.org or @neilalexander:matrix.org
with a link to the respective Github issue, thanks!
-->

**Comment by nakoo** [comment]
I can confirm that it only happens when using global.database. I tested with separated database and it doesn't have that issue.

**Comment by spaetz** [comment]
So, have you deleted all the database: blocks other than the global one?

**Comment by nakoo** [comment]
> So, have you deleted all the database: blocks other than the global one?

Yes, purged all the databases and dendrite data and started with pure `global.database` setup.

**Comment by soda-pop-ice-cream** [comment]
I have same issue. Had to enable registration to create accounts.

### Background information
- **Dendrite version or git SHA**: 0.8.4
- **Monolith or Polylith?**: Monolith
- **SQLite3 or Postgres?**: Postgres
- **Running in Docker?**: No
- **`go version`**: go1.18.1 linux/arm64
- **Client used (if applicable)**: Element Web
